src/components/implementation/tests/micro_booter/micro_booter.h

```diff
@@ -9,49 +9,80 @@
 
 #undef assert
 /* On assert, immediately switch to the "exit" thread */
-#define assert(node)                                          \
-	do {                                                  \
-		if (unlikely(!(node))) {                      \
-			debug_print("assert error in @ ");    \
-			cos_thd_switch(termthd[cos_cpuid()]); \
-		}                                             \
-	} while (0)
+#define assert(node)                                            \
+    do {                                                        \
+        if (unlikely(!(node))) {                                \
+            debug_print("assert error in @ ");                  \
+            cos_thd_switch(termthd[cos_cpuid()]);               \
+        }                                                       \
+    } while (0)
+
+#define EXIT_FN()                                               \
```

Comment:
This feels far too clever to me.  I think the explicit declaration of the lab, and the use of `COS_THROW` is better.

Reply:
I have no good answer atm, I don't remember my logic to using this. 

Nvm, it exists because of thd needing to communicate that they failed that's the reason. The idea if I remember correctly was that if a thread (not the main one) encountered an error, it needed to communicate the failure. That's the purpose of `CHECK_STATUS_FLAG` but I'll rename it   COS_THROW.

The macro for the label, I do not remember why it was that way. The idea came after a talk to @phanikishoreg